kerykeion/void_of_course_moon/utils.py

```
@@ -0,0 +1,204 @@
+    jd = jd0 + guess_days
+    for _ in range(_MAX_ITERATIONS):
+        longitude, speed = _lon_speed(jd, swe.MOON, iflag)
+        error = difdeg2n(longitude, target_longitude)  # signed degrees, 0 at crossing
+        if not speed or abs(error) < _ANGLE_EPSILON:
+            break
+        jd -= error / speed
+    return jd
+
+
+def _aspect_perfection_jd(jd_guess: float, body_id: int, signed_target: float, iflag: int) -> Optional[float]:
+    """Newton-refine the Julian Day where Moon−body separation equals ``signed_target``."""
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)